corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -182,26 +185,55 @@ hqDefine("geospatial/js/gps_capture",[
         self.saveDataRow = function (dataItem) {
             self.isSubmissionSuccess(false);
             self.hasSubmissionError(false);
+            let dataItemJson = ko.mapping.toJS(dataItem);
+            if (self.createCaseEnabled()) {
+                const caseType = $('#report_filter_case_type').val();
+                dataItemJson['case_type'] = caseType;
+            }
+
             $.ajax({
                 method: 'POST',
                 url: initialPageData.reverse('gps_capture'),
                 data: JSON.stringify({
                     'data_type': self.dataType,
-                    'data_item': ko.mapping.toJS(dataItem),
+                    'data_item': dataItemJson,
+                    'create_case': self.createCaseEnabled(),
                 }),
                 dataType: "json",
                 contentType: "application/json; charset=utf-8",
                 success: function () {
                     dataItem.hasUnsavedChanges(false);
                     self.isSubmissionSuccess(true);
                     resetMap();
+                    self.createCaseEnabled(false);
                 },
                 error: function () {
                     self.hasSubmissionError(true);
                 },
             });
         };
 
+        self.startCreateCase = function () {
+            self.createCaseEnabled(true);
+            const caseToCreate = new dataItemModel(null, self.dataType);
+            self.captureLocationForItem(caseToCreate);
+        };
+
+        self.finishCreateCase = function () {
+            const hasValidName = self.itemLocationBeingCapturedOnMap().name().length > 0;
+            self.createCaseError(!hasValidName);
+            if (!hasValidName) {
+                return;
+            }
+            self.saveDataRow(self.itemLocationBeingCapturedOnMap());
+        };
+
+        self.cancelCreateCase = function () {
+            self.createCaseEnabled(false);
+            self.createCaseError(false);
```

Comment:
nit: may be you could extract a function `resetCaseCreate` that resets the flags `createCaseError` and `createCaseEnabled` since this could be used at different places like after successful case creation.